### PR TITLE
Fix/improvements michiel

### PR DESCRIPTION
**Pull request by Michiel-s**
Combination of different improvements in several components
Work of last weekend.

### CODE REVIEW

File: src/app/admin/installer/installer.component.ts [hunk 1]
```diff
@@ -1,19 +1,55 @@
 import { HttpClient } from '@angular/common/http';
 import { Component } from '@angular/core';
+import { finalize } from 'rxjs';
+
+class ButtonState {
+  loading: boolean = false;
```
Comment (LiWeiYeh) [inline src/app/admin/installer/installer.component.ts:6]: missing data modifiers

File: src/app/admin/installer/installer.component.ts [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
+  buttonState1 = new ButtonState();
+  buttonState2 = new ButtonState();
+  buttonState3 = new ButtonState();
+
   constructor(private http: HttpClient) {}
 
-  reinstall(): void {
-    this.http.get('admin/installer').subscribe();
+  isLoading(): boolean {
+    return [this.buttonState1, this.buttonState2, this.buttonState3].some(state => state.isLoading());
```
Comment (LiWeiYeh) [inline src/app/admin/installer/installer.component.ts:38]: I think it's better if you split this up in a variable name that describes what is happening, e.g. `let isButtonLoading: boolean = [this.buttonState1, this.buttonState2, this.buttonState3].some(state => state.isLoading());` and then `return isButtonLoading`

File: src/app/app.component.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { LayoutService } from './layout/service/app.layout.service';
 
 @Component({
   selector: 'app-root',
-  templateUrl: './app.component.html',
+  template: '<router-outlet></router-outlet>',
```
Comment (LiWeiYeh) [inline src/app/app.component.ts:8]: In my opinion, I think it's still better to have a `app.component.html` page and reference that through the template in the `.ts` file to have the consistency as other angular component files

**Review by LiWeiYeh — CHANGES_REQUESTED** [review]
mostly a request for change of styling, but looks good!

File: src/app/admin/installer/installer.component.ts (hunk 1 above)
Comment (Michiel-s) [reply src/app/admin/installer/installer.component.ts:6]: what do you mean?

File: src/app/admin/installer/installer.component.ts (hunk 2 above)
Comment (Michiel-s) [reply src/app/admin/installer/installer.component.ts:38]: Accept for having a 'isButtonLoading' in between, I don't see the point. Btw it is not the button that is loading, but the action/call to backend triggered by the button.

I'll add a comment to the function

File: src/app/app.component.ts (hunk 3 above)
Comment (Michiel-s) [reply src/app/app.component.ts:8]: I understand and agree (in general). In this case the AppComponent is a special root/starting component that in our case is nothing more that `<router-outlet>`.

File: src/app/admin/installer/installer.component.ts [hunk 4]
```diff
@@ -1,19 +1,55 @@
...
+  }
+}
 
 @Component({
   selector: 'app-installer',
   templateUrl: './installer.component.html',
   styleUrls: ['./installer.component.scss'],
 })
 export class InstallerComponent {
+  buttonState1 = new ButtonState();
```
[the remark this answers is not in the document]
Comment (Michiel-s) [reply src/app/admin/installer/installer.component.ts:31]: Check. I'll add te ButtonState type

File: src/app/admin/installer/installer.component.ts (hunk 1 above)
Comment (Michiel-s) [reply src/app/admin/installer/installer.component.ts:6]: Like public/private? Ok. I'll add that.

**Review by LiWeiYeh — APPROVED** [review]

**Pull request merged**